products/mlengine/api.yaml

```diff
@@ -235,4 +235,338 @@ objects:
+           where both the key and the value are arbitrary Strings that you supply. 
+           An object containing a list of "key": value pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+      # no etag since it's input: true     
+      ## trainingInput starts here     
+      - !ruby/object:Api::Type::NestedObject
+        name: 'trainingInput'
+        description: Input parameters to create a training job.
+        output: false
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'scaleTier'
+            description: Specifies the machine types, the number of replicas for workers and parameter servers.
+            values:
+              - BASIC
+              - STANDARD_1
+              - PREMIUM_1
+              - BASIC_GPU
+              - BASIC_TPU
+              - CUSTOM
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'masterType'
+            description: Specifies the type of virtual machine to use for your training job's master worker  
+            required: false  
+          - !ruby/object:Api::Type::NestedObject
+            name: 'masterConfig'
+            description: The configuration for your master worker.
+            properties:
+              - !ruby/object:Api::Type::NestedObject
+                name: 'acceleratorConfig'
+                description: Represents the type and number of accelerators used by the replica
+                properties:
+                - !ruby/object:Api::Type::Integer
+                  name: 'count'
+                  description: The number of accelerators to attach to each machine running the job.
+                - !ruby/object:Api::Type::Enum
+                  name: 'type'
+                  description: The type of accelerator to use.
+                  values:
+                    - ACCELERATOR_TYPE_UNSPECIFIED
+                    - NVIDIA_TESLA_K80	
+                    - NVIDIA_TESLA_P100
+                    - NVIDIA_TESLA_V100
+                    - NVIDIA_TESLA_P4
+                    - NVIDIA_TESLA_T4
+                    - TPU_V2
+                    - TPU_V3 
+              - !ruby/object:Api::Type::String
+                name: 'imageUri'
+                description: The Docker image to run on the replica. This image must be in Container Registry.
+              - !ruby/object:Api::Type::String
+                name: 'tpuTfVersion'
+                description: |
+                  The AI Platform runtime version that includes a TensorFlow version matching the one used in the 
+                  custom container. This field is required if the replica is a TPU worker that uses a custom container. 
+                  Otherwise, do not specify this field.
+          - !ruby/object:Api::Type::String
+            name: 'workerType'
+            description: Specifies the type of virtual machine to use for your training job's worker nodes.
```

Comment:
It looks like there's some further description in the GCP docs, can you add that in? (here and elsewhere)

Reply:
added